src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -755,25 +760,25 @@ private static bool IsPotentialFilterCharacter(char c)
 
         private static (int index, bool forceHardSelection) GetDefaultsMatch(
             string filterText,
-            ImmutableArray<(RoslynCompletionItem item, PatternMatch? patterMatch)> itemsWithMatch,
+            List<MatchResult<VSCompletionItem>> itemsWithMatch,
```

Comment:
why a List?

Reply:
This is a transient object backed by a pooled.